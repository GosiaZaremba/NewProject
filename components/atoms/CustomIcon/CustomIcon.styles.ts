import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  pressable: {
    opacity: 1,
  },
  pressed: {
    opacity: 0.5,
  },
  iconFB: {
    fontSize: 24,
  },
});
