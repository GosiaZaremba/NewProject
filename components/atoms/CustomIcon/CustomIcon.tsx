import React from 'react';
import {Pressable} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import styles from './CustomIcon.styles';

export type Props = {
  onPressIcon: () => void;
  iconColor: string;
  iconName: string;
};

export const CustomIcon: React.FC<Props> = ({
  onPressIcon,
  iconColor,
  iconName,
}) => {
  return (
    <Pressable
      onPress={onPressIcon}
      hitSlop={5}
      style={({pressed}) => (pressed ? styles.pressed : styles.pressable)}>
      <Icon name={iconName} style={styles.iconFB} color={iconColor}></Icon>
    </Pressable>
  );
};
