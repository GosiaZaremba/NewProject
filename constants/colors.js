export const Colors = {
  pink: '#fc296f',
  red: '#e63039',
  navy: '#39539b',
  blue: '#3872c3',
  grey: '#908D8E',
  light: '#F5F0F2',
};
