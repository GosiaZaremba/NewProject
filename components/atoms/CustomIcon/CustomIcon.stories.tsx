/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { CustomIcon } from './CustomIcon';
import { Colors } from '../../../constants/colors';
import { color, number, text } from '@storybook/addon-knobs';

storiesOf('atoms/CustomIcon', module)
    .addDecorator((story) => story())
    .add('default', () => (
        <CustomIcon
            iconColor={color('blue', Colors.blue)}
            iconName={text('name', 'facebook')}
            iconSize={number('fontSize', 18)}
            iconBorderColor={color('blue', Colors.blue)}
            onPressIcon={() => {}}
            testOnly_pressed={true}
        />
    ));
