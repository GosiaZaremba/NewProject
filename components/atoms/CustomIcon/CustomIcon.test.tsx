import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CustomIcon } from './CustomIcon';
import { Colors } from '../../../constants/colors';

const mockOnPress = jest.fn();

describe('Atom CustomIcon', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <CustomIcon
                    iconColor={Colors.blue}
                    iconName={'facebook'}
                    iconSize={18}
                    iconBorderColor={Colors.blue}
                    onPressIcon={mockOnPress}
                    testOnly_pressed={true}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure icon works correctly', () => {
        const { getByTestId } = render(
            <CustomIcon
                iconColor={Colors.blue}
                iconName={'facebook'}
                iconSize={18}
                iconBorderColor={Colors.blue}
                onPressIcon={mockOnPress}
                testID={'atom-icon'}
                testOnly_pressed={true}
            />
        );
        const customIcon = getByTestId('atom-icon-icon');
        fireEvent.press(customIcon);
        expect(mockOnPress).toHaveBeenCalledTimes(1);
        expect(customIcon.props.style[1].fontSize).toBe(18);
        expect(customIcon.props.style[0].color).toBe(Colors.blue);
        // console.log(customIcon.props);
    });
    test('ensure pressable works correctly when pressed', () => {
        const { getByTestId } = render(
            <CustomIcon
                iconColor={Colors.blue}
                iconName={'facebook'}
                iconSize={18}
                iconBorderColor={Colors.blue}
                onPressIcon={mockOnPress}
                testID={'atom-icon'}
                testOnly_pressed={true}
            />
        );
        const iconPressable = getByTestId('atom-icon-pressable');
        fireEvent.press(iconPressable);
        expect(mockOnPress).toHaveBeenCalledTimes(2);
        expect(iconPressable.props.style[1].borderColor).toBe(Colors.blue);
        expect(iconPressable.props.style[0].opacity).toBe(0.5);
    });

    test('ensure pressable works correctly', () => {
        const { getByTestId } = render(
            <CustomIcon
                iconColor={Colors.blue}
                iconName={'facebook'}
                iconSize={18}
                iconBorderColor={Colors.blue}
                onPressIcon={mockOnPress}
                testID={'atom-icon'}
                testOnly_pressed={false}
            />
        );
        const iconPressable = getByTestId('atom-icon-pressable');
        fireEvent.press(iconPressable);
        expect(mockOnPress).toHaveBeenCalledTimes(3);
        // expect(iconPressable.props.style[1].borderColor).toBe(Colors.blue);
        // expect(iconPressable.props.style[0].opacity).toBe(undefined);
        console.log(iconPressable.props.style);
    });
});
