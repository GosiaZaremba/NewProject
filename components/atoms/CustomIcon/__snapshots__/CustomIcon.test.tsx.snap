// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom CustomIcon Component Snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={5}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "borderWidth": 3,
        "height": 45,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 0.5,
        "padding": 10,
        "width": 45,
      },
      Object {
        "borderColor": "#3872c3",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#3872c3",
          "fontSize": 12,
        },
        Object {
          "fontSize": 18,
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï‚š
  </Text>
</View>
`;
